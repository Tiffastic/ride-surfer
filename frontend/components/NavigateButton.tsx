import React from "react";
import { Button, Linking, View } from "react-native";

import Colors from "../constants/Colors";
import Styles from "../constants/Styles";

// https://developers.google.com/maps/documentation/urls/guide#directions-action

const url = "https://www.google.com/maps/dir/?api=1&travelmode=walking";

export default function NavigateButton(props: { dest: string }) {
  return (
    <View style={Styles.buttonView}>
      <Button
        title="Navigate"
        onPress={() =>
          Linking.openURL(`${url}&destination=${encodeURI(props.dest)}`)
        }
        color={Colors.primary}
      />
    </View>
  );
}
