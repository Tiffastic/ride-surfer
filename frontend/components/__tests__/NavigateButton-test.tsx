import "react-native";
import React from "react";
import NavigateButton from "../NavigateButton";
import renderer from "react-test-renderer";

it("renders correctly", () => {
  const tree = renderer
    .create(<NavigateButton dest="Salt Lake City, UT" />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
