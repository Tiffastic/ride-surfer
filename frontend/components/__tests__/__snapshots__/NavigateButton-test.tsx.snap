// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "fontSize": 24,
      "marginVertical": 5,
    }
  }
>
  <View
    accessibilityRole="button"
    accessibilityStates={Array []}
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
            Object {
              "color": "#194cc0",
            },
          ]
        }
      >
        Navigate
      </Text>
    </View>
  </View>
</View>
`;
